### PR TITLE
change trim spec to be case insensitive 

### PR DESCRIPTION
**Pull request by yliuuuu**
*Issue #, if available:*
Closes Issue #744 

*Description of changes:*
1. Added support for trim to accept case insensitive [ BOTH | LEADING | TRAILING]
2. Fixed error message in case of Trim Spec mismatch
3. Added an Parser Error to specifically for trim spec mismatch
4. Added test case for upper case trim spec
5. Added test cases for Trim Spec Mismatch

Future to do: 
Decide whether or not Trim Spec mismatch shall be a parser error or an evaluator error, and clean up the code accordingly. 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lang/test/org/partiql/lang/errors/ParserErrorsTest.kt [hunk 1]
```diff
@@ -1286,6 +1286,36 @@ class ParserErrorsTest : SqlParserTestBase() {
         )
     }
 
+    @Test
+    fun callTrimSpecificationMismatch() {
+        // for SQL parser, the logic seems to be if the first token is not one of { NONE | BOTH | LEADING | TRAILING}
+        // treating the token as identifier and expect the trim expression to end
```
Comment (lziq) [inline lang/test/org/partiql/lang/errors/ParserErrorsTest.kt:1292]: Can you add a `TODO` here saying that this test case will be removed once we remove the legacy `SqlParser`?

File: lang/src/org/partiql/lang/errors/ErrorCode.kt [hunk 2]
```diff
@@ -531,6 +531,12 @@ enum class ErrorCode(
         "Other expressions may not be present in the select list when '*' is used without dot notation."
     ),
 
+    PARSE_INVALID_ARGUMENTS_FOR_TRIM(
```
Comment (lziq) [inline lang/src/org/partiql/lang/errors/ErrorCode.kt:534]: Probably change the name to `PARSE_INVALID_TRIM_SPEC`? The term "arguments" can refer to the string being trimmed.

**Review by lziq — CHANGES_REQUESTED** [review]
Overall looks good to me. I think trim specification mismatch should be considered as a parser error.

File: lang/src/org/partiql/lang/errors/ErrorCode.kt (hunk 2 above)
Comment (yliuuuu) [reply lang/src/org/partiql/lang/errors/ErrorCode.kt]: Addressed in the new commit.

File: lang/test/org/partiql/lang/errors/ParserErrorsTest.kt (hunk 1 above)
Comment (yliuuuu) [reply lang/test/org/partiql/lang/errors/ParserErrorsTest.kt:1293]: Addressed in the new commit. I will open up another issue on treating specification mismatch as a parser error.

**Review by lziq — DISMISSED** [review]
As what we discussed offline, `trim` spec mismatch error will be considered further in another PR.

**Pull request merged**